4890 mixmax triggers and actions

luancazarine opened the pull request:
Mixmax triggers and actions

Triggers:
  - New Contact
  - New Contact Group

Actions:
  - Get Contacts
  - Create Contact
  - Update Contact
  - Search Contacts
  - Get Contact Groups
  - Create Contact Group

feyzullah commented on components/mixmax/sources/common/base-single.mjs:
```diff
@@ -0,0 +1,50 @@
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
```
Do we need this file here at all?

feyzullah commented on components/mixmax/package.json:
```diff
@@ -12,5 +12,8 @@
     "access": "public"
+  },
+  "dependencies": {
+    "@pipedream/platform": "^1.1.1"
```
I guess pnpm will still use `1.2.1` but can we change it `1.2.1` as it is the latest version?
```suggestion
    "@pipedream/platform": "^1.2.1"
```

feyzullah commented on pnpm-lock.yaml:
```diff
@@ -2837,6 +2840,14 @@ importers:
   components/teamdeck:
     specifiers: {}
 
+  components/teamgate:
```
I guess `Teamgate`'s dependencies were also added here and it seems it uses `@pipedream/platform` version `0.10.0`. If possible can we omit these in this PR?

feyzullah reviewed (changes requested):
Hi @luancazarine, I added a few comments. Can you please check?

feyzullah reviewed (dismissed):
LGTM!

DilanAthukorala commented:
@luancazarine Observing following issues,


1. Create Contact Action: Groups are not listed
![image](https://user-images.githubusercontent.com/102980146/207419841-d9ea8f44-99ba-4797-a7d2-19b0628119c6.png)


2. Create Group Action: Contact Id's are not passed to MixMax
![image](https://user-images.githubusercontent.com/102980146/207425492-fa65a886-6511-4cfd-bb4b-020bbe86696b.png)


3. Get Groups action doesn't return Groups, it only returns contacts. 


4. Get Contacts Action throws error
![image](https://user-images.githubusercontent.com/102980146/207427211-c4252d62-46c9-4083-a5e1-8a7066f78d63.png)


5. Search Contacts Throws error when sending request without "q" String
![image](https://user-images.githubusercontent.com/102980146/207428428-4fb2f482-426b-47e3-a94a-c9e1937f41ec.png)

luancazarine commented:
Hello @DilanAthukorala, About the issues:

1. Create Contact Action: Groups are not listed
Did you create the groups? it was supposed to be working.
![image](https://user-images.githubusercontent.com/5638699/207697516-9e4b3f86-3f6a-4f19-889e-3afbe0f65d45.png)

2. Create Group Action: Contact Id's are not passed to MixMax
Fixed!

3. Get Groups action doesn't return Groups, it only returns contacts.
Fixed!

4. Get Contacts Action throws error
Fixed! 

5. Search Contacts Throws error when sending request without "q" String
This error comes from the API so I made it required

DilanAthukorala commented:
@luancazarine For get contact groups action - error throws when sending request without fields. 
![image](https://user-images.githubusercontent.com/102980146/207936520-26b75705-2e58-4b76-b522-3b2e1ec0a95e.png)

DilanAthukorala commented:
@luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?

luancazarine commented:
> @luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?

@DilanAthukorala 
![image](https://user-images.githubusercontent.com/5638699/208673134-0869fad5-cd58-4fd9-a01d-a5eb646b23a3.png)

DilanAthukorala commented:
> > @luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?
> 
> @DilanAthukorala ![image](https://user-images.githubusercontent.com/5638699/208673134-0869fad5-cd58-4fd9-a01d-a5eb646b23a3.png)

@luancazarine You have to select at least one group to observe the issue

DilanAthukorala commented:
Verification completed, ready for release.

luancazarine commented:
/approve

lcaresia reviewed (approved):
LGTM!
